pyteal/ast/block_test.py

```diff
@@ -58,3 +60,37 @@ def test_block_timestamp():
 def test_block_timestamp_invalid():
     with pytest.raises(pt.TealTypeError):
         pt.Block.timestamp(pt.Txn.sender())
+
+
+def test_block_v11_fields():
+    arg = pt.Int(0)
+    cases = [
+        ("proposer", "BlkProposer", pt.TealType.bytes),
+        ("fees_collected", "BlkFeesCollected", pt.TealType.uint64),
+        ("bonus", "BlkBonus", pt.TealType.uint64),
+        ("branch", "BlkBranch", pt.TealType.bytes),
+        ("fee_sink", "BlkFeeSink", pt.TealType.bytes),
+        ("protocol", "BlkProtocol", pt.TealType.bytes),
+        ("txn_counter", "BlkTxnCounter", pt.TealType.uint64),
+        ("proposer_payout", "BlkProposerPayout", pt.TealType.uint64),
+    ]
+    for tc in cases:
+        block_method = getattr(pt.Block, tc[0])
+        expr = block_method(arg)
+        assert expr.type_of() == tc[2]
+
+        expected = pt.TealSimpleBlock(
+            [
+                pt.TealOp(arg, pt.Op.int, 0),
+                pt.TealOp(expr, pt.Op.block, tc[1]),
+            ]
+        )
+
+    actual, _ = expr.__teal__(avm11Options)
+    actual.addIncoming()
+    actual = pt.TealBlock.NormalizeBlocks(actual)
+
+    assert actual == expected
+
+    with pytest.raises(pt.TealInputError):
+        expr.__teal__(avm10Options)
```

Comment:
Why raise the options from previous version (v10) here?

Reply:
It's saying that if you call `expr._teal_` and specify that you want it compiled for v10, then an exception will be raised (since the field didn't exist in v10).